include/zephyr/sys/util.h

```diff
@@ -546,6 +546,35 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 		(expr);                                                                            \
 	})
 
+/** @brief Calculate the base-2 logarithm */
+#define LOG2(x) (sizeof(uint64_t) * 8 - __builtin_clzll((uint64_t)(x)-1))
```

Comment:
What is the behaviour here with smaller types? The majority of Zephyrs targets are 32bit instead of 64bit AFAIK, so it seems likely that the usual use case would be calling this on something <= 32 bits. It seems likely this is quite inefficient.